<launch>
  <arg name="UI_TYPE" default="mouse" />
  <arg name="USE_ROTATION" default="false" />
  <arg name="POSITION_THRESHOLD" default="0.01" /> <!-- meter -->
  <arg name="ROTATION_THRESHOLD" default="2" /> <!-- degree -->
  <arg name="MAKE_INTERACTIVE_MARKER_ARROW" default="true" />
  <arg name="LAUNCH_RVIZ_WITH_DEFAULT_CONFIG" default="true" />

  <group if="$(arg LAUNCH_RVIZ_WITH_DEFAULT_CONFIG)">
    <node pkg="rviz" type="rviz" name="interactive_test_rviz"
	  args="-d $(find jsk_interactive_test)/launch/rviz.rviz" respawn="true"/>
  </group>

  <node pkg="jsk_interactive_test" name="publish_goal_marker"
        type="publish_goal_marker.py" output="screen" clear_params="true">
    <param name="ui_type" value="$(arg UI_TYPE)" />
    <param name="use_rotation" value="$(arg USE_ROTATION)" />
    <param name="position_threshold" value="$(arg POSITION_THRESHOLD)" />
    <param name="rotation_threshold" value="$(arg ROTATION_THRESHOLD)" />
  <param name="make_interactive_marker_arrow" type="bool" value="$(arg MAKE_INTERACTIVE_MARKER_ARROW)" />  
  </node>
</launch>
